--- llvm_android/builders.py	2021-11-03 05:33:42.948074724 +0800
+++ ndk-aarch64-linux/llvm_android/builders.py	2021-11-03 05:34:23.248074709 +0800
@@ -155,7 +155,7 @@ class Stage2Builder(base_builders.LLVMBu
         env = super().env
         # Point CMake to the libc++ from stage1.  It is possible that once built,
         # the newly-built libc++ may override this because of the rpath pointing to
-        # $ORIGIN/../lib64.  That'd be fine because both libraries are built from
+        # $ORIGIN/../lib.  That'd be fine because both libraries are built from
         # the same sources.
         env['LD_LIBRARY_PATH'] = str(self.toolchain.lib_dir)
         return env
@@ -295,8 +295,10 @@ class BuiltinsBuilder(base_builders.LLVM
         sarch = 'i686' if arch == hosts.Arch.I386 else arch.value
         filename = 'libclang_rt.builtins-' + sarch + '-android.a'
         filename_exported = 'libclang_rt.builtins-' + sarch + '-android-exported.a'
-        src_path = self.output_dir / 'lib' / 'linux' / filename
 
+        src_path = self.output_dir / 'lib' / 'linux' / filename
+        print(src_path)
+        print(self.output_toolchain.resource_dir / filename)
         if self.is_exported:
             # This special copy exports its symbols and is only intended for use
             # in Bionic's libc.so.
@@ -446,6 +448,49 @@ class CompilerRTHostI386Builder(base_bui
         super()._build_config()
 
 
+class CompilerRTHostAarch64Builder(base_builders.LLVMRuntimeBuilder):
+    name: str = 'compiler-rt-aarch64-host'
+    src_dir: Path = paths.LLVM_PATH / 'compiler-rt'
+    config_list: List[configs.Config] = [configs.LinuxConfig(is_32_bit=False)]
+
+    @property
+    def install_dir(self) -> Path:
+        return self.output_toolchain.clang_lib_dir
+
+    @property
+    def cmake_defines(self) -> Dict[str, str]:
+        defines = super().cmake_defines
+        # Due to CMake and Clang oddities, we need to explicitly set
+        # CMAKE_C_COMPILER_TARGET and use march=i686 in cflags below instead of
+        # relying on auto-detection from the Compiler-rt CMake files.
+        defines['CMAKE_C_COMPILER_TARGET'] = 'aarch64-linux-gnu'
+        defines['COMPILER_RT_INCLUDE_TESTS'] = 'ON'
+        defines['COMPILER_RT_ENABLE_WERROR'] = 'ON'
+        defines['SANITIZER_CXX_ABI'] = 'libstdc++'
+        return defines
+
+    @property
+    def cflags(self) -> List[str]:
+        cflags = super().cflags
+        # compiler-rt/lib/gwp_asan uses PRIu64 and similar format-specifier macros.
+        # Add __STDC_FORMAT_MACROS so their definition gets included from
+        # inttypes.h.  This explicit flag is only needed here.  64-bit host runtimes
+        # are built in stage1/stage2 and get it from the LLVM CMake configuration.
+        # These are defined unconditionaly in bionic and newer glibc
+        # (https://sourceware.org/git/gitweb.cgi?p=glibc.git;h=1ef74943ce2f114c78b215af57c2ccc72ccdb0b7)
+        cflags.append('-D__STDC_FORMAT_MACROS')
+        cflags.append('--target=aarch64-linux-gnu')
+        cflags.append('-march=armv8-a')
+        return cflags
+
+    def _build_config(self) -> None:
+        # Also remove the "stamps" created for the libcxx included in libfuzzer so
+        # CMake runs the configure again (after the cmake caches are deleted).
+        stamp_path = self.output_dir / 'lib' / 'fuzzer' / 'libcxx_fuzzer_aarch64-stamps'
+        if stamp_path.exists():
+            shutil.rmtree(stamp_path)
+        super()._build_config()
+
 class LibUnwindBuilder(base_builders.LLVMRuntimeBuilder):
     name: str = 'libunwind'
     src_dir: Path = paths.LLVM_PATH / 'libunwind'
@@ -857,7 +902,7 @@ class SysrootsBuilder(base_builders.Buil
             # Create a stub library for the platform's libc++.
             platform_stubs = paths.OUT_DIR / 'platform_stubs' / config.ndk_arch
             platform_stubs.mkdir(parents=True, exist_ok=True)
-            libdir = sysroot / 'usr' / ('lib64' if arch == hosts.Arch.X86_64 else 'lib')
+            libdir = sysroot / 'usr' / ('lib' if arch == hosts.Arch.X86_64 else 'lib')
             libdir.mkdir(parents=True, exist_ok=True)
             with (platform_stubs / 'libc++.c').open('w') as f:
                 f.write(textwrap.dedent("""\
@@ -908,7 +953,7 @@ class PlatformLibcxxAbiBuilder(base_buil
 
     def install_config(self) -> None:
         arch = self._config.target_arch
-        lib_name = 'lib64' if arch == hosts.Arch.X86_64 else 'lib'
+        lib_name = 'lib' if arch == hosts.Arch.X86_64 else 'lib'
         install_dir = self._config.sysroot / 'usr' / lib_name
 
         if self._is_64bit():
@@ -943,7 +988,7 @@ class LibCxxBuilder(base_builders.LLVMRu
         # into install_dir only during libcxx's install step.  But use the
         # library from install_dir.
         defines['LIBCXX_CXX_ABI_INCLUDE_PATHS'] = str(paths.LLVM_PATH / 'libcxxabi' / 'include')
-        defines['LIBCXX_CXX_ABI_LIBRARY_PATH'] = str(self.libcxx_abi_path / 'lib64')
+        defines['LIBCXX_CXX_ABI_LIBRARY_PATH'] = str(self.libcxx_abi_path / 'lib')
 
         # Build only the static library.
         defines['LIBCXX_ENABLE_SHARED'] = 'OFF'
@@ -1039,7 +1084,7 @@ class WindowsToolchainBuilder(base_build
         ldflags.append(libpath_prefix + str(paths.WIN_ZLIB_LIB_PATH))
         if self.libcxx_path:
             # Add path to libc++, libc++abi.
-            libcxx_lib = self.libcxx_path / 'lib64'
+            libcxx_lib = self.libcxx_path / 'lib'
             ldflags.append(libpath_prefix + str(libcxx_lib))
         return ldflags
 
